gather/prometheus.go

```diff
@@ -21,7 +22,11 @@ type prometheusScraper struct{}
 
 // Gather parse metrics from a scraper target url.
 func (p *prometheusScraper) Gather(ctx context.Context, target influxdb.ScraperTarget) (collected MetricsCollection, err error) {
-	resp, err := http.Get(target.URL)
+	customTransport := http.DefaultTransport.(*http.Transport).Clone()
+	customTransport.TLSClientConfig = &tls.Config{InsecureSkipVerify: target.AllowInsecure}
+	client := &http.Client{Transport: customTransport}
```

Comment:
Ideally we wouldn't be creating a client every time. I think if you add an http client to the `prometheusScraper` with `InsecureSkipVerify` set to true then in this function, you could do something like:
```go
var (
    resp *http.Response
    err error
)

if target.AllowInsecure {
    resp, err = p.insecureClient.Get(target.URL)
} else {
    resp, err = http.Get(target.URL)
}
```

Reply:
Done!